request-manager/request/manager.go

```diff
@@ -540,39 +540,46 @@ func (m *manager) GetWithJC(requestId string) (proto.Request, error) {
 
 func (m *manager) Find(filter proto.RequestFilter) ([]proto.Request, error) {
 	// Build the query from the filter.
-	query := "SELECT request_id, type, state, user, created_at, started_at, finished_at, total_jobs, finished_jobs, jr_url FROM requests "
+	query := "SELECT request_id, type, state, user, created_at, started_at, finished_at, total_jobs, finished_jobs, jr_url" +
+		" FROM requests r LEFT JOIN request_archives ra USING (request_id) "
 
 	var fields []string
 	var values []interface{}
 	if filter.Type != "" {
-		fields = append(fields, "type = ?")
+		fields = append(fields, "r.type = ?")
 		values = append(values, filter.Type)
 	}
 	if filter.User != "" {
-		fields = append(fields, "user = ?")
+		fields = append(fields, "r.user = ?")
 		values = append(values, filter.User)
 	}
 	if len(filter.States) != 0 {
-		stateSQL := fmt.Sprintf("state IN (%s)", strings.TrimRight(strings.Repeat("?, ", len(filter.States)), ", "))
+		stateSQL := fmt.Sprintf("r.state IN (%s)", strings.TrimRight(strings.Repeat("?, ", len(filter.States)), ", "))
 		fields = append(fields, stateSQL)
 		for _, state := range filter.States {
 			values = append(values, state)
 		}
 	}
+	if len(filter.Args) != 0 {
+		for arg, val := range filter.Args {
+			fields = append(fields, "ra.create_request LIKE CONCAT('%\"', ?, '\":\"', ?, '\"%')")
+			values = append(values, arg, val)
+		}
+	}
```

Comment:
Could you add a test for the args filter to [TestFind](https://github.com/square/spincycle/blob/cf48fe75816ae2e0932844dd8069e1757329d2d9/request-manager/request/manager_test.go#L662)?

Reply:
I was looking to add a test for this but it seems that the existing test framework does not use/populate the `request_archives` table. So args will always be empty.

Reply:
I'm not sure how to update it to populate `request_archives` table